evaluation.py

```diff
@@ -237,7 +237,143 @@ def predict_keypoints(self,generator, location):
+                    right_hip_point = np.array(annotation_keypoints[36], annotation_keypoints[37])
+                    torso = np.linalg.norm(left_hip_point-right_hip_point)
+                    threshold = PCK_THRESHOLD*torso
+
+                    for i in range(NUM_COCO_KEYPOINTS):
+                        base = i*3
+                        prediction_point = np.array(prediction_keypoints[base], prediction_keypoints[base+1])
+                        annotation_point = np.array(annotation_keypoints[base], annotation_keypoints[base+1])
+                        dist = (np.linalg.norm(prediction_point-annotation_point))
+                        dist_list.append(dist)
+
+            # Each image may have more than one annotation we need to check
+            annotations = int(len(dist_list)/NUM_COCO_KEYPOINTS)
+
+            nose_correct = False
+            left_eye_correct = False
+            right_eye_correct = False
+            left_ear_correct = False
+            right_ear_correct = False
+            left_shoulder_correct = False
+            right_shoulder_correct = False
+            left_elbow_correct = False
+            right_elbow_correct = False
+            left_wrist_correct = False
+            right_wrist_correct = False
+            left_hip_correct = False
+            right_hip_correct = False
+            left_knee_correct = False
+            right_knee_correct = False
+            left_ankle_correct = False
+            right_ankle_correct = False
+
+            # Append True to correct joint list if distance is below threshold for any annotation
+            for j in range(annotations):
+                base = j*NUM_COCO_KEYPOINTS
+                if dist_list[0+base]<=threshold: nose_correct = True
+                if dist_list[1+base]<=threshold: left_eye_correct = True
+                if dist_list[2+base]<=threshold: right_eye_correct = True
+                if dist_list[3+base]<=threshold: left_ear_correct = True
+                if dist_list[4+base]<=threshold: right_ear_correct = True
+                if dist_list[5+base]<=threshold: left_shoulder_correct = True
+                if dist_list[6+base]<=threshold: right_shoulder_correct = True
+                if dist_list[7+base]<=threshold: left_elbow_correct = True
+                if dist_list[8+base]<=threshold: right_elbow_correct = True
+                if dist_list[9+base]<=threshold: left_wrist_correct = True
+                if dist_list[10+base]<=threshold: right_wrist_correct = True
+                if dist_list[11+base]<=threshold: left_hip_correct = True
+                if dist_list[12+base]<=threshold: right_hip_correct = True
+                if dist_list[13+base]<=threshold: left_knee_correct = True
+                if dist_list[14+base]<=threshold: right_knee_correct = True
+                if dist_list[15+base]<=threshold: left_ankle_correct = True
+                if dist_list[16+base]<=threshold: right_ankle_correct = True
```

Comment:
this would do the same thing (and is cleaner), but I'm good either way!


```suggestion
                nose_correct            = nose_correct              or dist_list[0+base]<=threshold
                left_eye_correct        = left_eye_correct          or dist_list[1+base]<=threshold
                right_eye_correct       = right_eye_correct         or dist_list[2+base]<=threshold
                left_ear_correct        = left_ear_correct          or dist_list[3+base]<=threshold
                right_ear_correct       = right_ear_correct         or dist_list[4+base]<=threshold
                left_shoulder_correct   = left_shoulder_correct     or dist_list[5+base]<=threshold
                right_shoulder_correct  = right_shoulder_correct    or dist_list[6+base]<=threshold
                left_elbow_correct      = left_elbow_correct        or dist_list[7+base]<=threshold
                right_elbow_correct     = right_elbow_correct       or dist_list[8+base]<=threshold
                left_wrist_correct      = left_wrist_correct        or dist_list[9+base]<=threshold
                right_wrist_correct     = right_wrist_correct       or dist_list[10+base]<=threshold
                left_hip_correct        = left_hip_correct          or dist_list[11+base]<=threshold
                right_hip_correct       = right_hip_correct         or dist_list[12+base]<=threshold
                left_knee_correct       = left_knee_correct         or dist_list[13+base]<=threshold
                right_knee_correct      = right_knee_correct        or dist_list[14+base]<=threshold
                left_ankle_correct      = left_ankle_correct        or dist_list[15+base]<=threshold
                right_ankle_correct     = right_ankle_correct       or dist_list[16+base]<=threshold
```